module/skill/QuinticWalk/src/QuinticWalk.cpp

```diff
@@ -0,0 +1,382 @@
+                + current_cfg.params.trunk_pitch_p_coef_turn * std::abs(walk_engine.get_footstep().get_next().z());
+            RPY.y() += wanted_pitch;
+
+            // threshold pitch and roll
+            if (std::abs(RPY.x()) > current_cfg.imu_roll_threshold) {
+                log<NUClear::WARN>(fmt::format("Robot roll exceeds threshold - {} > {}",
+                                               std::abs(RPY.x()),
+                                               current_cfg.imu_roll_threshold));
+                walk_engine.request_pause();
+            }
+            else if (std::abs(RPY.y()) > current_cfg.imu_pitch_threshold) {
+                log<NUClear::WARN>(fmt::format("Robot pitch exceeds threshold - {} > {}",
+                                               std::abs(RPY.y()),
+                                               current_cfg.imu_pitch_threshold));
+                walk_engine.request_pause();
+            }
+        });
+
+        on<Configuration>("QuinticWalk.yaml").then([this](const Configuration& config) {
+            log_level = config["log_level"].as<NUClear::LogLevel>();
+
+            load_quintic_walk(config, normal_config);
+
+            // Make sure the walk engine has the parameters at least once
+            if (first_config) {
+                // Send these parameters to the walk engine
+                walk_engine.set_parameters(current_cfg.params);
+
+                imu_reaction.enable(current_cfg.imu_active);
+
+                first_config = false;
+            }
+        });
+
+        on<Configuration>("goalie/QuinticWalk.yaml").then([this](const Configuration& config) {
+            load_quintic_walk(config, goalie_config);
+        });
+
+        on<Startup, Trigger<KinematicsModel>>().then("Update Kinematics Model", [this](const KinematicsModel& model) {
+            kinematicsModel = model;
+            first_run       = true;
+            current_orders.setZero();
+            is_left_support  = true;
+            last_update_time = NUClear::clock::now();
+            walk_engine.reset();
+        });
+
+        // Runs every time the Walk provider starts (wasn't running)
+        on<Start<Walk>, With<Behaviour::State>>().then([this](const Behaviour::State& behaviour) {
+            walk_engine.reset();
+            if (behaviour == Behaviour::State::GOALIE_WALK) {
+                current_config = goalie_config;
+            }
+            else {
+                current_config = normal_config;
+            }
+            // Send these parameters to the walk engine
+            walk_engine.set_parameters(current_cfg.params);
+
+            imu_reaction.enable(current_cfg.imu_active);
+        });
+
+        // Runs every time the Walk task is removed from the director tree
+        on<Stop<Walk>>().then([this] {
+            const float dt = get_time_delta();
+            current_orders.setZero();
+            walk_engine.update_state(dt, current_orders);
```

Comment:
These lines should be removed (given you reset everything in `Start<Walk>`). Keep the `imu_reaction` line in though.